Classes/Fluid/ViewHelper/ComponentRenderer.php

```diff
@@ -257,19 +302,26 @@ public function compile(
      * @param \Closure $renderChildrenClosure
      * @param RenderingContextInterface $renderingContext
      * @param string $componentNamespace
+     * @param ComponentContext|null $componentContext overwrite component context
      * @return mixed
      */
     public static function renderComponent(
         array $arguments,
         \Closure $renderChildrenClosure,
         RenderingContextInterface $renderingContext,
-        $componentNamespace
+        $componentNamespace,
+        ?ComponentContext $componentContext = null
     ) {
         $viewHelperClassName = get_called_class();
 
         $objectManager = GeneralUtility::makeInstance(ObjectManager::class);
         $componentRenderer = $objectManager->get($viewHelperClassName);
         $componentRenderer->setComponentNamespace($componentNamespace);
+        if (isset($componentContext)) {
+            $componentRenderer->setComponentContext($componentContext);
+        } else {
+            $componentRenderer->getComponentContext()->reset();
```

Comment:
Maybe this can be optimized so that the merging and copying is only necessary in case context is actually in use?